specs-go/chain/chainid.go

```diff
@@ -0,0 +1,46 @@
+package chain
+
+import "github.com/docker/distribution/digest"
+
+// ChainID takes a slice of digests and returns the ChainID corresponding to
+// the last entry. Typically, these are a list of layer DiffIDs, with the
+// result providing the ChainID identifying the result of sequential
+// application of the preceeding layers.
+func ChainID(dgsts []digest.Digest) digest.Digest {
+	chainIDs := make([]digest.Digest, len(dgsts))
+	copy(chainIDs, dgsts)
+	ChainIDs(chainIDs)
+
+	if len(chainIDs) == 0 {
+		return ""
```

Comment:
And with all of the proposals in #482, the chainID for an empty array is *still* undefined.  In the absence of a spec definition (which is where things stand with PR), I think we should be raising an “undefined” error.